From ad3ac5d5696e0490fe28ea09fb8067545c14a34a Mon Sep 17 00:00:00 2001
From: Daiki Mizukami <tesaguriguma@gmail.com>
Date: Tue, 10 Oct 2023 07:23:14 +0900
Subject: [PATCH] login: Make `login -f` work on volumes with the nosuid option
 set

---
 login.tproj/login.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/login.tproj/login.c b/login.tproj/login.c
index d32a06d..51018f5 100644
--- a/login.tproj/login.c
+++ b/login.tproj/login.c
@@ -869,7 +869,7 @@ main(int argc, char *argv[])
 	 * We don't need to be root anymore, so set the login name and
 	 * the UID.
 	 */
-	if (setlogin(username) != 0) {
+	if (uid == (uid_t)0 && setlogin(username) != 0) {
 		syslog(LOG_ERR, "setlogin(%s): %m - exiting", username);
 		bail(NO_SLEEP_EXIT, 1);
 	}
-- 
2.37.3

